docs/src/examples/Supervised-NN-ODE-MNIST.md

```diff
@@ -94,24 +88,280 @@ function accuracy(model,data; n_batches=100)
+`Chain` allows inputs to functions to come from the previous layer and send the outputs
+to the next. Four different sets of layers are used here.
+
+
+```julia
+down = Chain(x->reshape(x,(28*28,:)),
+             Dense(784,20,tanh)
+            ) |> gpu
+
+nn  = Chain(Dense(20,10,tanh),
+            Dense(10,10,tanh),
+            Dense(10,20,tanh)
+           ) |> gpu
+
+
+nn_ode = NeuralODE(nn, (0.f0, 1.f0), Tsit5(),
+                        save_everystep = false,
+                        reltol = 1e-3, abstol = 1e-3,
+                        save_start = false) |> gpu
+
+fc  = Chain(Dense(20,10)) |> gpu
+
+nfe = 0
+```
+
+`down`: Takes a 28x28 image, flattens it, then passes it through a fully connected
+layer with `tanh` activation<br>
+`nn`: Neural network chain three layers deep, `tanh` activation <br>
+`nn_ode`: ODE solver for neural network <mark><-- Should add more explanation </mark><br>
+`fc`: Final fully connected layer <br>
+`nfe`: <mark><-- not sure </mark><br>
+`|> gpu`: sends to GPU
+
+### Array Conversion
+
+Cheap conversion of `DiffEqArray` from ODE solver into Matrix.
+
+```julia
+function DiffEqArray_to_Array(x)
+    xarr = gpu(x)
+    return reshape(xarr, size(xarr)[1:2])
+end
+```
+
+This is required to convert the output of the Neural Network ODE.
+
+### Build Topology
+
+Connect all layers together in a single chain
+
+```julia
+#Build our over-all model topology
+m = Chain(down,
+          nn_ode,
+          DiffEqArray_to_Array,
+          fc) |> gpu
+
+#To understand the intermediate NN-ODE layer, we can examine it's dimensionality
+x_d = down(x_train[1])
+
+#We can see that we can compute the forward pass through the NN topology featuring an NNODE layer.
+x_m = m(x_train[1])
+```
+
+This can also be built without the NN-ODE by replacing nn-ode with a simple nn.
+
+```julia
+#We can also build the model topology without a NN-ODE
+m_no_ode = Chain(down,
+                 nn,
+                 fc) |> gpu
+
+x_m = m_no_ode(x_train[1])
+```
+
+### Prediction
+
+Function that returns the prediction by taking the arg max of the output for each column
+of the minibatch.
+
+```julia
+classify(x) = argmax.(eachcol(x))
+```
+
+### Accuracy
+
+Check accuracy by running the network through 100 batches at a time.
+Use the `classify` function created above
+
+```julia
+function accuracy(model,data; n_batches=100)
+    total_correct = 0
+    total = 0
+    for (x,y) in collect(data)[1:n_batches]
+        target_class = classify(cpu(y))
+        predicted_class = classify(cpu(model(x)))
+        total_correct += sum(target_class .== predicted_class)
+        total += length(target_class)
+    end
+    return total_correct/total
+end
+
+#burn in accuracy
+accuracy(m, zip(x_train,y_train))
+```
+
+### Loss
+```julia
+loss(x,y) = logitcrossentropy(m(x),y)
+
+#burn in loss
+loss(x_train[1],y_train[1])
+```
+
+### Optimizer
+```julia
+opt = ADAM(0.05)
+iter = 0
+```
+
+### <mark>Unsure</mark>
```

Comment:
this is a callback for printing the loss during the training. It prints the current accuracy on the training set every 10 iterations by updating `iter` as a global each step and then doing `@show accuracy...`

Reply:
That makes sense. What is nfe used for here?

Reply:
number of function evaluations. That's used to count the number of times that the callback is called so it only prints the loss every 10th time.